mesa/space.py

```diff
@@ -56,9 +56,8 @@
 
 
 def clamp(x: float, lowest: float, highest: float) -> float:
-    # This should be faster than np.clip for a scalar x.
-    # TODO: measure how much faster this function is.
-    return max(lowest, min(x, highest))
+    # much faster than np.clip for a scalar x.
+    return lowest if x <= lowest else (highest if x >= highest else x)
```

Comment:
Are you sure about the `<=` and `>=` sign? Looks wrong to me.

Reply:
I'm pretty sure about it, when you have `x `lower than `lowest` it returns  `lowest` as in the original function and so on...

Reply:
Oh nvm, looks correct. However, it probably needs to be like this?
```python
x = highest if x >= highest else x
# with the new x
return lowest if x <= lowest else x
```

Reply:
mmmh...I think that the order of the operations doesn't matter if `highest >= lowest`. But in some way the other case should be treated in a special way it could be done your way, what do you think about this?

Reply:
anyway timing your alternative it goes on about the same speed `c = timeit.timeit(stmt="clamp(0.3, 0.0 ,1.0)", setup='def clamp(x: float, lowest: float, highest: float) -> float: \n\tx = highest if x >= highest else x\n\t# with the new x\n\treturn lowest if x <= lowest else x')`

Reply:
Your implementation is better. If you test on arbitrary 3 numbers, sometimes it is faster, because when `x <= lowest` is true, the second if-else block doesn't even get evaluated, which saves time.

Reply:
Now you just need to:
1. rebase properly
2. squash your commits
3. properly write your commit message according to Tim Pope's guideline (see CONTRIBUTING.rst)

Anyway, this is all already specified in CONTRIBUTING.rst.

Reply:
thanks, I'm working on that!